weblate/trans/views/basic.py

```diff
@@ -351,11 +361,61 @@ def new_language(request, project, component):
     else:
         form = form_class(obj)
 
-    return render(
-        request,
-        "new-language.html",
-        {"object": obj, "project": obj.project, "form": form, "can_add": can_add},
+    context = {
+        "object": obj,
+        "project": obj.project,
+        "form": form,
+        "can_add": can_add,
+        "namespace": namespace,
+        "namespaced_form": NewNamespacedLanguageForm(obj, namespace=namespace),
+    }
+    return render(request, "new-language.html", context)
+
+
+@never_cache
+@login_required
+def new_namespaced_language(request, project, component):
+    obj = get_component(request, project, component)
+
+    namespace = ""
+    namespace_query = request.user.groups.filter(roles__name=ACCESS_NAMESPACE).order_by(
+        "name"
     )
+    if namespace_query.count():
+        namespace = namespace_query[0]
```

Comment:
Is it ever possible for a user to be in multiple partner groups? Do super admins need access to create namespaced translations on behalf of a partner?

Reply:
I believe it's technically possible, but only for internal users. We may need to create namespaced translations on behalf of a partner, and perhaps we can use query parameters for that when the time comes.